core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowFlinkPlugin.scala

```diff
@@ -0,0 +1,240 @@
+      // pack all streamlet-descriptors into a Json array
+      val streamletDescriptorsJsonStr =
+        streamletsInProject
+          .value
+          .toJson
+          .compactPrint
+
+      val compressed = zlibCompression(streamletDescriptorsJsonStr.getBytes(UTF_8))
+      val streamletDescriptorsLabelValue =
+        Base64.getEncoder.encodeToString(compressed)
+
+      new Dockerfile {
+        from(CloudflowFlinkDockerBaseImage)
+        user(userInImage)
+
+        copy(depJarsDir, optAppDir, chown = userAsOwner(userInImage))
+        copy(appJarsDir, optAppDir, chown = userAsOwner(userInImage))
+        runRaw(s"cp ${optAppDir}cloudflow-runner.jar  /opt/flink/flink-web-upload/cloudflow-runner.jar")
+        label(streamletDescriptorsLabelName, streamletDescriptorsLabelValue)
+      }
+    },
+
+    build := showResultOfBuild.dependsOn(
+      docker.dependsOn(
+        checkUncommittedChanges
+      )
+    ).value,
+
+    buildAndPublish := showResultOfBuildAndPublish.dependsOn(
+      dockerBuildAndPush.dependsOn(
+        checkUncommittedChanges,
+        verifyDockerRegistry
+      )
+    ).value,
+
+    fork in Compile := true
+  )
+
+  private val verifyDockerRegistry = Def.task {
+    cloudflowDockerRegistry.value.getOrElse(throw DockerRegistryNotSet)
+  }
+
+  private val checkUncommittedChanges = Def.task {
+    val log = streams.value.log
+    if (cloudflowBuildNumber.value.hasUncommittedChanges) {
+      log.warn(s"You have uncommitted changes in ${thisProjectRef.value.project}. Please commit all changes before publishing to guarantee a repeatable and traceable build.")
+    }
+  }
+
+  private val showResultOfBuild = Def.task {
+    val log = streams.value.log
+    val imagePushed = (imageNames in docker).value.head // assuming we only build a single image!
+
+    log.info(" ") // if you remove the space, the empty line will be auto-removed by SBT somehow...
+    log.info("Successfully built the following Cloudflow application image:")
+    log.info(" ")
+    log.info(s"  $imagePushed")
+    log.info(" ")
+    log.info("Before you can deploy this image to a Kubernetes cluster you will")
+    log.info("need to push it to a docker registry that is reachable from the cluster.")
+    log.info(" ")
+  }
+
+  private val showResultOfBuildAndPublish = Def.task {
+    val log = streams.value.log
+    val imagePushed = (imageNames in docker).value.head // assuming we only build a single image!
+
+    log.info(" ") // if you remove the space, the empty line will be auto-removed by SBT somehow...
+    log.info("Successfully built and published the following Cloudflow application image:")
+    log.info(" ")
+    log.info(s"  $imagePushed")
+    log.info(" ")
+    log.info("You can deploy the application to a Kubernetes cluster using any of the the following commands:")
+    log.info(" ")
+    log.info(s"  kubectl cloudflow deploy $imagePushed")
```

Comment:
This is incorrect. We can choose to already write here what we want it to be (deploying the blueprint, ```kubectl cloudflow <blueprint-file>` ), which means we must also add in the text that the user has to update the blueprint file.

Maybe it is best to just remove this, and only say "Successfully built image:" for now.

Reply:
done ..